molgenis-data-security/src/main/java/org/molgenis/data/security/audit/EntityTypeRepositoryAuditDecorator.java

```diff
@@ -0,0 +1,88 @@
+package org.molgenis.data.security.audit;
+
+import static com.google.common.collect.Streams.stream;
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.data.security.audit.AuthenticationUtils.getUsername;
+import static org.molgenis.data.semantic.Relation.isAudited;
+import static org.molgenis.data.semantic.Vocabulary.AUDIT_USAGE;
+
+import java.util.HashMap;
+import java.util.stream.Stream;
+import org.molgenis.audit.AuditEvent;
+import org.molgenis.audit.AuditEventPublisher;
+import org.molgenis.data.AbstractRepositoryDecorator;
+import org.molgenis.data.Repository;
+import org.molgenis.data.meta.model.EntityType;
+import org.molgenis.data.meta.model.Tag;
+
+/** Publishes {@link AuditEvent}s when an {@link EntityType}'s auditing is enabled or disabled. */
+public class EntityTypeRepositoryAuditDecorator extends AbstractRepositoryDecorator<EntityType> {
+
+  static final String AUDIT_ENABLED = "AUDIT_ENABLED";
+  static final String AUDIT_DISABLED = "AUDIT_DISABLED";
+
+  private final AuditEventPublisher auditEventPublisher;
+
+  public EntityTypeRepositoryAuditDecorator(
+      Repository<EntityType> delegateRepository, AuditEventPublisher auditEventPublisher) {
+    super(delegateRepository);
+    this.auditEventPublisher = requireNonNull(auditEventPublisher);
+  }
+
+  @Override
+  public void update(EntityType entityType) {
+    auditAuditTagChanges(entityType);
+    delegate().update(entityType);
+  }
+
+  @Override
+  public void update(Stream<EntityType> entities) {
+    delegate().update(entities.filter(this::auditAuditTagChanges));
+  }
+
+  @Override
+  public void add(EntityType entityType) {
+    auditAuditTag(entityType);
+    delegate().add(entityType);
+  }
+
+  @Override
+  public Integer add(Stream<EntityType> entities) {
+    return delegate().add(entities.filter(this::auditAuditTag));
```

Comment:
If add logs the add, and delete does not log the removal, you can quietly delete and then re-add without a tag.